src/Cache/Invalidation.php

```diff
@@ -757,4 +762,151 @@ public function on_insert_comment_cb( $comment_id, $comment ) {
 			$this->purge( 'list:comment' );
 		}
 	}
+
+	/**
+	 * Given a key and nodes, dispatch an HTTP request to the
+	 * GraphQL endpoint to purge the specified caches
+	 *
+	 * @param $key
+	 * @param $nodes
+	 *
+	 * @return void
+	 * @throws Exception
+	 */
+	public function dispatch_purge_cache( $key, $nodes ) {
+
+		// if there are no nodes to purge, bail
+		if ( empty( $nodes ) ) {
+			return;
+		}
+
+		$query = '
+		mutation AsyncPurgeGraphqlCache( $input: AsyncPurgeGraphqlCacheInput! ) {
+		  asyncPurgeGraphqlCache( input: $input ) {
+		    success
+		    queryIds
+		    key
+		  }
+		}
+		';
+
+		$variables = [
+			'input' => [
+				'clientMutationId' => uniqid( 'graphql_purge', true ),
+				'key' => $key,
+				'queryIds' => $nodes,
+				'nonce' => wp_create_nonce( 'graphql_purge' )
+			],
+		];
+
+		$purge_async = apply_filters( 'graphql_smart_cache_purge_async', true );
+
+		// if purge async is disabled, purge internally
+		if ( ! $purge_async ) {
+
+			// if async purging is disabled,
+			// use the graphql() function internally
+			// to purge the cache.
+			// This is used for unit testing, and can be used
+			// in certain debug scenarios as well.
+			graphql([
+				'query' => $query,
+				'variables' => $variables,
+			]);
+		} else {
+
+			// get the current cookies and prepare them to
+			// be sent in the remote POST request
+			// so it can execute as the user
+			// making the current request.
+			$cookies = [];
+			foreach ( $_COOKIE as $name => $value ) {
+				$cookies[] = new \WP_Http_Cookie([
+					'name' => $name,
+					'value' => $value,
+				]);
+			}
+
+			// send a post request to the GraphQL endpoint
+			wp_safe_remote_post( site_url( Router::$route ), [
+				'headers'  => [
+					'Content-Type' => 'application/json',
+				],
+				'body'     => wp_json_encode( [
+					'query'     => $query,
+					'variables' => $variables,
+				] ),
+				// this is a non-blocking HTTP request so that
+				// the current action causing cache evictions
+				// can complete independent of the caches being
+				// looped over and evicted.
+				'blocking' => false,
```

Comment:
The only thing that `'blocking' => false` saves is the parsing of the response body. It still waits for HTTP response in order to avoid hanging up:

https://core.trac.wordpress.org/browser/trunk/src/wp-includes/class-requests.php#L632

Therefore, this request will still wait for purges to complete, now with the added overhead of another HTTP request / WordPress bootup. :/

True async will require something like `cron` or an external queue.

Reply:
@chriszarate I confirmed this by putting a `sleep( 10 );` in the `mutateAndGetPayload` of the `asyncPurgeGraphqlCache` mutation. 

However, I also confirmed that setting a super low timeout on the request gets back to the behavior I was hoping for.